### DISCUSSION TITLE
Dismiss and Alert ends with strange behaviour

### DISCUSSION THREAD

**Discussion by mpiotrowskibsg** [pointfreeco/swift-composable-architecture#1993] [Q&A]
### Description

It seems that there is issue with presenting and dismissing in new navigation API. Checking Integration app and new Presentation API I can see issue where I use `Dismiss and Alert`. It ends with just dismissing and console error. It also makes next presentation broken because it is presented and automatically dismissed.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

1. Presented screen is dismissed
2. Alert is shown

### Actual behavior

1. Presented screen is dismissed
2. Error is printed on cosole
3. Next presentation is broken

### Steps to reproduce

1. Open Integration app
2. Select Presentations APIs
3. Press Open Sheet
4. Press Dismiss and Alert
5. Press again Open Sheet to see next broken presentation

### The Composable Architecture version information

prerelease/1.0

### Destination operating system

iOS 16.1

### Xcode version information

14.1

### Swift Compiler version information

```shell
swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx12.0
```

**Comment by stephencelis** [comment]
@mpiotrowskibsg Our integration suite doesn't have a ton of documentation at the moment, but it shouldn't be viewed as "case studies" of working examples. It doesn't just exercise happy paths, but also exercises "unhappy" paths that sometime demonstrate some unfortunate vanilla SwiftUI behavior. In this case, SwiftUI itself just doesn't support programmatically swapping a sheet with an alert. The workaround most folks seem to apply is to space out dismissal and presentation using delays.

I'm going to convert this issue to a discussion since it's not a bug with the library, but rather a current behavior of vanilla SwiftUI.

**Reply by mbrandonw** [reply]
And we even have a test that demonstrates the bug, and so hopefully someday this test will start passing:

https://github.com/pointfreeco/swift-composable-architecture/blob/bbdfd264a4ac24fa98968c1433217e242cff78e5/Examples/Integration/IntegrationUITests/PresentationTests.swift#L315-L324
